### ISSUE TITLE
Ensure that MsgCreateValidator's accompanying delegation is at least the provided MinSelfDelegation

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/cosmos-sdk#112]
## Summary

We should ensure that on MsgCreateValidator, if the self-delegation sent in the message isn't at least the provided MinSelfDelegation parameter, no new validator object is created.

## Problem Definition

This, accompanied with a global minimum self delegation puts a non-zero cost barrier to spamming validator counts, which can cause notable complexity in other users codebase.

## Proposal

Add a check right here: https://github.com/osmosis-labs/cosmos-sdk/blob/f4329814320de3f0f6170f9347053c28afc60831/x/staking/keeper/msg_server.go#L52-L57 . 

Basically just do `msg.Value.Amount.GTE(msg.MinSelfDelegation)`. If not, return an error.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
